community/kernel/src/main/java/org/neo4j/kernel/impl/api/MandatoryPropertyEnforcer.java

```diff
@@ -117,15 +173,29 @@ private PrimitiveIntIterator labelsOf( long nodeId )
         }
     }
 
-    private Iterator<PropertyConstraint> mandatoryPropertyConstraints( int label )
+    private Iterator<NodePropertyConstraint> mandatoryNodePropertyConstraints( int label )
     {
-        return new FilteringIterator<>( storeLayer.constraintsGetForLabel( label ), new Predicate<PropertyConstraint>()
-        {
-            @Override
-            public boolean test( PropertyConstraint constraint )
-            {
-                return constraint instanceof MandatoryPropertyConstraint;
-            }
-        } );
+        return new FilteringIterator<>( storeLayer.constraintsGetForLabel( label ),
+                new Predicate<NodePropertyConstraint>()
```

Comment:
Or better yet, make this entire predicate object a constant, since we don't need to re-instantiate it for each call to `mandatoryNodePropertyConstraints`.